packages/fxa-auth-server/scripts/verification-reminders.js

```diff
@@ -126,7 +135,77 @@ async function run() {
     }
   }, Promise.resolve());
 
+  // TODO: This is intentionally an exact copy of the above code
+  // since CAD reminders are an experiment. At the end we'll either
+  // refactor reminders into a better abstraction of remove this code.
+  await cadReminders.keys.reduce(async (promise, key) => {
+    await promise;
+
+    const method = `cadReminder${key[0].toUpperCase()}${key.substr(1)}Email`;
+    const reminders = cReminders[key];
+
+    log.info('cadReminders.processing', {
+      count: reminders.length,
+      key,
+    });
+
+    const failedReminders = await reminders.reduce(
+      async (promise, { timestamp, uid, flowId, flowBeginTime }) => {
+        const failed = await promise;
+
+        try {
+          if (sent[uid]) {
+            // Don't send e.g. first and second reminders to the same email from a single batch
+            log.info('cadReminders.skipped.alreadySent', { uid });
+            failed.push({ timestamp, uid, flowId, flowBeginTime });
+            return failed;
+          }
+
+          const account = await db.account(uid);
+          await mailer[method]({
+            acceptLanguage: account.locale,
+            code: account.emailCode,
+            email: account.email,
+            flowBeginTime,
+            flowId,
+            uid,
+          });
+          // eslint-disable-next-line require-atomic-updates
+          sent[uid] = true;
+        } catch (err) {
+          const { errno } = err;
+          switch (errno) {
+            case error.ERRNO.ACCOUNT_UNKNOWN:
+            case error.ERRNO.BOUNCE_COMPLAINT:
+            case error.ERRNO.BOUNCE_HARD:
+            case error.ERRNO.BOUNCE_SOFT:
+              log.info('cadReminders.skipped.error', { uid, errno });
+              try {
+                await cadReminders.delete(uid);
+              } catch (ignore) {}
+              break;
+            default:
+              log.error('cadReminders.error', { err });
+              failed.push({ timestamp, uid, flowId, flowBeginTime });
+          }
+        }
+
+        return failed;
+      },
+      Promise.resolve([])
+    );
+
+    if (failedReminders.length > 0) {
+      log.info('cadReminders.reinstating', {
+        count: reminders.length,
+        key,
+      });
+      return cadReminders.reinstate(key, failedReminders);
```

Comment:
The cad reminders library doesn't have the reinstate call. I'm not sure how valuable this is in practice, do we need it?

Reply:
i'll remove it